daemon/tests/happy_path.rs

```diff
@@ -76,13 +73,13 @@ async fn taker_takes_order_and_maker_accepts_and_contract_setup() {
     taker.take_order(received.clone(), Usd::new(dec!(5)));
     let (_, _) = next_cfd(&mut taker.cfd_feed, &mut maker.cfd_feed).await;
 
-    mock_oracle_annoucement(&mut maker.mocks.oracle().await);
-    mock_oracle_annoucement(&mut taker.mocks.oracle().await);
+    maker.mocks.mock_oracle_annoucement().await;
```

Comment:
In an ideal world, we can even get rid of the `.mocks` indirection without too much duplication!

Reply:
I don't mind this small indirection as it enforces separation of concerns, as it gets rid of any duplication (and there will be more methods...)